contracts/ligo/src/minter_collection/editions/fa2_multi_nft_token_editions.mligo

```diff
@@ -0,0 +1,122 @@
+    let edition_metadata : edition_metadata = {
+      creator = Tezos.sender;
+      edition_info = param.edition_info;
+      initial_token_id = storage.nft_asset_storage.assets.next_token_id;
+      number_of_editions = param.number_of_editions;
+      number_of_editions_to_distribute = int(param.number_of_editions) 
+    } in
+    let new_editions_metadata = Big_map.add storage.current_edition_id edition_metadata storage.editions_metadata in
+    let new_asset_storage = {storage.nft_asset_storage.assets with next_token_id = storage.nft_asset_storage.assets.next_token_id + param.number_of_editions} in 
+    {storage with 
+        current_edition_id = storage.current_edition_id + 1n;
+        editions_metadata = new_editions_metadata; 
+        nft_asset_storage = {storage.nft_asset_storage with assets = new_asset_storage}
+    } in
+  (match List.head_opt editions_list with 
+    | Some editions -> let new_storage = mint_single_edition_set(editions, storage) in 
+        (match List.tail_opt editions_list with
+          | Some remaining_editions -> mint_editions (remaining_editions, new_storage)
+          | None -> (failwith "INTERNAL_ERROR" : operation list * editions_storage))
+    | None -> ([] : operation list), storage)
+
+let distribute_edition_to_address (edition_metadata, to_, storage : edition_metadata * address * editions_storage) 
+  : editions_storage * edition_metadata = 
+  let token_id = edition_metadata.initial_token_id + abs (edition_metadata.number_of_editions - edition_metadata.number_of_editions_to_distribute) in 
+  let new_edition_metadata = {edition_metadata with number_of_editions_to_distribute = edition_metadata.number_of_editions_to_distribute - 1} in
+  let mint_token_param : mint_token_param = {
+      token_metadata = {
+          token_id = token_id;
+          token_info = (Map.empty : (string, bytes) map); 
+      };
+      owner = to_;
+  } in
+  let mint_tokens_param : mint_tokens_param = [mint_token_param] in 
+  let _ , nft_token_storage = mint_tokens (mint_tokens_param, storage.nft_asset_storage.assets) in
+  {storage with nft_asset_storage = {storage.nft_asset_storage with assets = nft_token_storage}}, new_edition_metadata
+
+let rec distribute_edition_to_addresses ( receivers, edition_metadata, storage : (address list) * edition_metadata * editions_storage)
+  : editions_storage * edition_metadata = 
+  match (List.head_opt receivers) with 
+    | Some to_ -> let new_storage, new_edition_metadata = distribute_edition_to_address (edition_metadata, to_, storage) in 
+        (match (List.tail_opt receivers) with 
+            | Some remaining_receivers -> distribute_edition_to_addresses (remaining_receivers, new_edition_metadata, new_storage)
+            | None -> (failwith "INTERNAL_ERROR" : editions_storage * edition_metadata))   
+    | None -> 
+        if (edition_metadata.number_of_editions_to_distribute < 0) then 
```

Comment:
Can this condition be checked only once at the end of the loop in `distribute_editions`?

Reply:
No because `distribute_editions` loops over multiple editions, so it would need to check the metadata for every edition it updated metadata for in the `editions_metadata` big map. I think this is the best we can do.